### PR TITLE
Group17 reachability refactoring

### PR DESCRIPTION
**Pull request by henneboy**
- Added comments
- Use BFS instead of DFS for theoretically better performance in larger transition systems (not big enough benchmarks to check sadly)

### CODE REVIEW

**Review by Darkros1245 — COMMENTED** [review]
lgtm

File: src/System/reachability.rs [hunk 1]
```diff
@@ -197,12 +211,12 @@ fn remove_existing_subsets_of_zone(
...
 fn make_path(mut sub_path: Rc<SubPath>) -> Path {
     let mut path: Vec<Transition> = Vec::new();
-
+    // Traverse the subpaths to make the path (from end location to start location)
     while sub_path.previous_sub_path.is_some() {
         path.push(sub_path.transition.clone().unwrap());
         sub_path = Rc::clone(sub_path.previous_sub_path.as_ref().unwrap());
     }
-
+    // Revese the path since the transitions are in reverse order (now from start location to end location)
```
Comment (Casper-NS) [inline src/System/reachability.rs:219]: little spelling mistake in "Revese"

Comment (henneboy) [reply src/System/reachability.rs]: Done 👍

**Review by Mightyhaha — APPROVED** [review]
Sure.

**Pull request merged**